### ISSUE TITLE
TPU running problem

### ISSUE THREAD

**Issue by circle1992** [googlecolab/colabtools#1814]
Hello, I have got a problem while using TPU for running the model training. My code runs okay using the GPU, but while changing it to TPU, it comes out the error as follows. Could someone help me with this?  Appreciate in advance.

---------------------------------------------------------------------------
UnavailableError                          Traceback (most recent call last)
<ipython-input-42-c040272eb558> in <module>()
    107 
    108 best = fmin(fn = objective, space = nnparams, algo = tpe.suggest, 
--> 109         max_evals = MAX_EVALS, trials = bayes_trials, verbose=2) 
    110 
    111 outfile = '/content/gdrive/My Drive/Colab Notebooks/BF/CSVcheckpoint/bestvalue'

13 frames
/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/context.py in sync_executors(self)
    643     """
    644     if self._context_handle:
--> 645       pywrap_tfe.TFE_ContextSyncExecutors(self._context_handle)
    646     else:
    647       raise ValueError("Context is not initialized.")

UnavailableError: {{function_node __inference_train_function_91768}} failed to connect to all addresses
Additional GRPC error information from remote target /job:localhost/replica:0/task:0/device:CPU:0:
:{"created":"@1610559945.714641633","description":"Failed to pick subchannel","file":"third_party/grpc/src/core/ext/filters/client_channel/client_channel.cc","file_line":4143,"referenced_errors":[{"created":"@1610559945.714638886","description":"failed to connect to all addresses","file":"third_party/grpc/src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc","file_line":398,"grpc_status":14}]}
	 [[{{node IteratorGetNext}}]]

**Comment by SivaPrasad02** [comment]
same problem did you find any solutions @circle1992

**Comment by the-cat-lady** [comment]
Same - I finished training running TPU, but when I went to save my trained model and assets, this error was thrown.  
My path was valid (I printed it out before the crash):

saving model to drive/MyDrive/data/output/bi_gru.pkl
INFO:tensorflow:Assets written to: drive/MyDrive/data/output/bi_gru.pkl/assets

And the containing directory was created in my drive.  But the directory is empty and the model is lost.
